lib/cli.nix

```diff
@@ -43,6 +43,15 @@ rec {
     options: attrs: lib.escapeShellArgs (toGNUCommandLine options attrs);
 
   toGNUCommandLine = {
+    # any preprocessing of an attribute value before we attempt
+    # to convert it to a command line option; by default we try
+    # to unwrap any value that appears to have been wrapped by
+    # mkForce/mkOverride.
+    #
+    # NOTE: If you override `mkOption` to support attrsets
+    # then you may need to override this attribute as well.
+    preprocessValue ? v: v?content then v.content else v,
```

Comment:
Maybe this should be another formatting option like `mkBool`. Since `mkOption` might have been the obvious choice, maybe something like `mkConfigurationOption` might work.

Reply:
Is 971980054ffc18ce1ffbaf9319c1dffdfec1e0f4 what you are looking for?